pkg/client/clientcmd/client_config.go

```diff
@@ -250,6 +250,9 @@ func (config DirectClientConfig) Namespace() (string, error) {
 	if len(configContext.Namespace) == 0 {
 		return api.NamespaceDefault, nil
 	}
+	if configContext.Namespace == "*" {
```

Comment:
I have a little confused and this may be wrong, "not storing * to namespace" means something like this:
```
if configContext.Namespace == "*" {
  configContext.AllNamespace = true
  configContext.Namespace = ""
}
if len(configContext.Namespace) == 0 {
  if configContext.NamespaceAll {
    return api.NamespaceAll, nil
  } else {
    return api.NamespaceDefault, nil
  }
}
```

So, which allow users to specifiy namespace as `*`, but need to clear namespace in context, right?